tests/unit_test/app_common/executors/task_script_runner_test.py

```diff
@@ -68,7 +72,159 @@ def test_app_scripts_with_sub_dirs3(self):
+        sys.path.append(os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/site-1/custom"))
+
+        try:
+            script_path = "failed_train.py"
+            wrapper = TaskScriptRunner(
+                site_name="site-1", script_path=script_path, script_args=script_args, redirect_print_to_log=False
+            )
+            wrapper.event_manager.data_bus.subscribe([TOPIC_ABORT, TOPIC_STOP], self.abort_callback)
+
+            self.assertTrue(wrapper.script_full_path.endswith(script_path))
+            with pytest.raises(ValueError, match="failed to train model"):
+                # 1 ) check if the exception is through,
+                # 2 ) more important to see if the callback is trigger.
+                wrapper.run()
+        finally:
+            sys.path = old_sys_path
+
+    def abort_callback(self, topic, data, databus):
+        print("\n ===== calling abort_callback begin")
+        # assert failure here will not cause test to fail
+        self.assertEqual(topic, TOPIC_ABORT)
+        print("\n ===== calling abort_callback end")
+
+    def test_run_relative_import_scripts(self):
+        old_sys_path = sys.path
+        script_args = "--batch_size 4"
+        # test the run should not throw exception for the relative path import.
+        sys.path.append(os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/custom"))
+        sys.path.append(os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/server/custom"))
+        sys.path.append(os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/site-1/custom"))
+
+        try:
+            script_path = "relative_import_train.py"
+            wrapper = TaskScriptRunner(
+                site_name="site-1", script_path=script_path, script_args=script_args, redirect_print_to_log=False
+            )
+            self.assertTrue(wrapper.script_full_path.endswith(script_path))
+            path = os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/site-1/custom")
+            msg = f"attempted relative import with no known parent package, the relative import is not support. python import is based off the sys.path: {path}"
+            with pytest.raises(ImportError, match=msg):
+                # check the ImportError
+                wrapper.run()
+        finally:
+            sys.path = old_sys_path
+
+    def test_run_abs_path_scripts(self):
+        old_sys_path = sys.path
+        script_args = "--batch_size 4"
+        # test the run should not throw exception for the relative path import.
+        sys.path.append(os.path.join(os.getcwd(), "tests/unit_test/data/jobs/in_proc_job/site-1/custom"))
+
+        try:
+            script_path = "/foo/dummy/train.py"
+
+            with pytest.raises(ValueError, match="script_path='/foo/dummy/train.py' not found"):
+                wrapper = TaskScriptRunner(
+                    site_name="site-1", script_path=script_path, script_args=script_args, redirect_print_to_log=False
```

Comment:
I don't know if this test is saying the run **should** throw exception or **should not** for the relative path import.

From the test logic, seems we expect it to throw an ImportError?